packages/core/useTitle/index.ts

```diff
@@ -1,30 +1,44 @@
-import type { MaybeRef, MaybeRefOrGetter, ReadonlyRefOrGetter } from '@vueuse/shared'
-import { toRef, toValue } from '@vueuse/shared'
+import type { Fn, MaybeRef, MaybeRefOrGetter, ReadonlyRefOrGetter } from '@vueuse/shared'
+import { toRef, toValue, tryOnBeforeUnmount } from '@vueuse/shared'
 import type { ComputedRef, Ref } from 'vue-demi'
 import { watch } from 'vue-demi'
 import { useMutationObserver } from '../useMutationObserver'
 import type { ConfigurableDocument } from '../_configurable'
 import { defaultDocument } from '../_configurable'
 
-export type UseTitleOptionsBase =
-{
+export type UseTitleOptionsBase = {
   /**
-   * Observe `document.title` changes using MutationObserve
-   * Cannot be used together with `titleTemplate` option.
-   *
-   * @default false
+   * Stop watching the title when unmounted
+   * @default true
    */
-  observe?: boolean
-}
-| {
+  stopOnUnmount?: boolean
+
   /**
-   * The template string to parse the title (e.g., '%s | My Website')
-   * Cannot be used together with `observe` option.
-   *
-   * @default '%s'
+   * Restore the original title when unmounted
+   * @param originTitle original title
+   * @returns restored title
    */
-  titleTemplate?: MaybeRef<string> | ((title: string) => string)
-}
+  restoreOnUnmount?: (originTitle: string) => string
```

Comment:
```suggestion
  restoreOnUnmount?: false | (originalTitle: string, currentTitle: string) => string | null | undefined
```

Should also allow bypassing this behavior.